graphql_compiler/cost_estimation/filter_selectivity_utils.py

```diff
@@ -358,38 +256,41 @@ def _get_selectivity_fraction_of_interval(interval, quantiles):
     of the whole domain.
 
     Args:
-        interval: IntegerInterval defining the range of values
-        quantiles: a sorted list of N values separating the values of the field into N-1
-                   groups of almost equal size. The first element of the list is the
+        interval: Interval[T] defining the range of values
+        quantiles: a sorted list of N values of type T separating the values of the field
+                   into N-1 groups of almost equal size. The first element of the list is the
                    smallest known value, and the last element is the largest known value.
                    The i-th element is a value greater than or equal to i/N of all present values.
                    N has to be at least 2.
 
     Returns:
         float, the fraction of the values contained in the interval.
     """
+    if interval.is_empty():
+        return 0
+
     if len(quantiles) < 2:
         raise AssertionError(u"Need at least 2 quantiles: {}".format(len(quantiles)))
     # Since we can't be sure the minimum observed value is the
     # actual minimum value, we treat values less than it as part
     # of the first quantile. That's why we drop the minimum and
     # maximum observed values from the quantile list.
     proper_quantiles = quantiles[1:-1]
-    domain_interval_size = len(proper_quantiles) + 1
+    domain_interval_size = float(len(proper_quantiles) + 1)
     if interval.lower_bound is None and interval.upper_bound is None:
         interval_size = domain_interval_size
     elif interval.lower_bound is None:
         upper_bound_quantile = bisect.bisect_left(proper_quantiles, interval.upper_bound)
-        interval_size = 0.5 + upper_bound_quantile
+        interval_size = 0.5 + float(upper_bound_quantile)
     elif interval.upper_bound is None:
         lower_bound_quantile = bisect.bisect_left(proper_quantiles, interval.lower_bound)
-        interval_size = 0.5 + len(proper_quantiles) - lower_bound_quantile
+        interval_size = 0.5 + float(len(proper_quantiles) - lower_bound_quantile)
     else:
         lower_bound_quantile = bisect.bisect_left(proper_quantiles, interval.lower_bound)
         upper_bound_quantile = bisect.bisect_left(proper_quantiles, interval.upper_bound)
         if lower_bound_quantile == upper_bound_quantile:
             # https://math.stackexchange.com/questions/195245/
-            interval_size = 1.0 / 3
+            interval_size = 1.0 / 3.0
```

Comment:
Let's add a bit more info here. Imagine that link stops working — we still want to have some idea of what to look for.